### PR TITLE
backport: Merge bitcoin#22423, 22096

### PR DESCRIPTION
**Pull request by vijaydasmp**
bitcoin backports

### CODE REVIEW

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 @knst @PastaPastaPasta requesting review

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK

File: test/functional/p2p_addrfetch.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
```
Comment (knst) [inline test/functional/p2p_addrfetch.py:1]: Consider to backport bitcoin/bitcoin#22568 in a follow-up PR. There are no any important/critical fix, just tests improvements, so, that's an out-of-scope of current PR

**Review by knst — COMMENTED** [review]
utACK 651fad5c01e74d268f46b27756a9ce4616427e67

**Review by knst — APPROVED** [review]

**Review by PastaPastaPasta — APPROVED** [review]
utACK 651fad5c01e74d268f46b27756a9ce4616427e67

**Comment by PastaPastaPasta** [comment]
22423 was already merged in https://github.com/dashpay/dash/pull/6021

**Pull request merged**